### ISSUE TITLE
Timing restriction for Datasucker

### ISSUE THREAD

**Issue by Frizzler** [mtgred/netrunner#5490]
Hello,

first of all, thank you for fixing this:
https://github.com/mtgred/netrunner/issues/5462

Another issue from me:
Is it possible to restrict the use of Datasucker's paid ability to be used only after any "when encountered" abilities on the ice have triggered?

To explain this more in detail:
The Corp has a rezzed Data Raven.
A Parasite with two tokens on it is installed on it.
Runner also has a Datasucker with two tokens.
Runner now runs and encounters this Data Raven.
As currently implemented, Runner can use the paid ability of Datasucker (which causes Data Raven to die) before Data Raven's "when encountered" ability resolves.
This is not correct, because according to the run flow chart the "when encountered" abilities go first (6.9.3.a in run flow chart), before Runner can use paid abilities on the ice (6.9.3b in run flow chart) (confirmed by Jade from NISEI rules team).

By the way, Femme Fatale's ability can be used before the "when encountered" abilities on the ice, because Femme Fatale's ability is considered to trigger at 6.9.3a. So if you change any code, Femme Fatale should remain that way.

**Comment by NoahTheDuke** [comment]
Sadly, we don't have any way to enforce this right now. This is one of the many parts of the engine where we are not fine-grained enough in our handling of the timing and priority rules. Maybe someday this will be better but not in the foreseeable future.

**Comment by NBKelly** [comment]
Closed as completed via #7669 (you can no longer fire abilities while other abilities, including the encounter ones, are resolving)
